File: internal/sast/fixtures/javascript/ssti_positive.js
// internal/sast/fixtures/javascript/ssti_positive.js
const express = require("express");
const handlebars = require("handlebars");

const app = express();

app.get("/greet", (req, res) => {
  const userTpl = "Hello {{name}}, " + req.query.suffix;
  const compiled = handlebars.compile(userTpl);
  res.send(compiled({ name: req.query.name }));
});

File: internal/sast/fixtures/csharp/Ssti_positive.cs
// internal/sast/fixtures/csharp/Ssti_positive.cs
using Microsoft.AspNetCore.Mvc;
using RazorEngineCore;

public class SstiController : Controller
{
    public IActionResult Render(string suffix)
    {
        var engine = new RazorEngine();
        var template = "@Model.Name " + suffix;
        var compiled = engine.Compile(template);
        return Content(compiled.Run(new { Name = "world" }));
    }
}
